libs/barista-components/filter-field/src/filter-field.ts

```diff
@@ -805,6 +825,222 @@ export class DtFilterField<T = any>
+      // For that, allowArrowKeyFocusOnTags retains getting to position 0 before focusing a tag element.
+      if (this.allowArrowKeyFocusOnTags || !this._inputValue.length) {
+        this._handleArrowKeyNavigation({ left: true });
+      } else {
+        this.allowArrowKeyFocusOnTags = true;
+      }
+    } else if (
+      keyCode === RIGHT_ARROW &&
+      this._inputEl.nativeElement.selectionStart === this._inputValue.length
+    ) {
+      // The input element moves the cursor on keydown but the navigation with arrow keys is on keyup,
+      // causing the cursor to jump over the last position in the input and focuses the first focusable tag element.
+      // For that, allowArrowKeyFocusOnTags retains getting the cursor to the last position before focusing a tag element.
+      if (this.allowArrowKeyFocusOnTags || !this._inputValue.length) {
+        this._handleArrowKeyNavigation({ right: true });
+      } else {
+        this.allowArrowKeyFocusOnTags = true;
+      }
+    }
+  }
+
+  /** @internal Handles the navigation through the list of tags when using arrow keys */
+  _handleArrowKeyNavigation(navigationEvent: {
+    /** Contains the currently focused tag from which the event was fired from */
+    tag?: DtFilterFieldTag;
+    /** Whether to step back */
+    left?: boolean;
+    /** Whether to step forward */
+    right?: boolean;
+  }): void {
+    const focusedTagIndex = this._tags
+      .toArray()
+      .findIndex((tag) => tag.editButton === navigationEvent.tag?.editButton);
+    const inputElementHasFocus =
+      document.activeElement === this._inputEl.nativeElement;
+    const inSplittedMode = !!this._suffixTagData.length;
+
+    // Step focus out of input element and into tag list and early exit
+    if (
+      inputElementHasFocus &&
+      navigationEvent.left &&
+      navigationEvent.tag === undefined
+    ) {
+      this._getNextFocusableTagButton(
+        this._tags.toArray()[this._prefixTagData.length - 1],
+        true,
+      ).nativeElement.focus();
+      if (inSplittedMode) {
+        this.inPrefix = true;
+      }
+      return;
+    } else if (
+      inSplittedMode &&
+      inputElementHasFocus &&
+      navigationEvent.right &&
+      navigationEvent.tag === undefined
+    ) {
+      this._tags
+        .toArray()
+        [this._prefixTagData.length].editButton.nativeElement.focus();
+      this.inSuffix = true;
+      return;
+    }
+
+    const firstElementHasFocus =
+      this._tags.first.editButton.nativeElement === document.activeElement;
+    const lastElementHasFocus =
+      this._getNextFocusableTagButton(this._tags.last, true).nativeElement ===
+      document.activeElement;
+    const editButtonHasFocus =
+      navigationEvent.tag?.editButton.nativeElement === document.activeElement;
+    const deleteButtonHasFocus =
+      navigationEvent.tag !== undefined &&
+      this._getNextFocusableTagButton(navigationEvent.tag!, true)
+        .nativeElement === document.activeElement;
```

Comment:
This could be moved into a function that returns the current focused element - only one can be focused ;) 
So e.g. return 'edit-button' or 'delete-button'